From 88894bcb680ed2954c6e3fd9265c7c9869da7ea9 Mon Sep 17 00:00:00 2001
From: Shengjiu Wang <shengjiu.wang@nxp.com>
Date: Fri, 8 Jun 2018 18:36:03 +0800
Subject: [PATCH 3962/5242] MLK-18578: hdmi_rx: wait longer time for audio
 info frame

commit  13fb1f05e0f8e92bb8f3fd22894855f66ee1c455 from
https://source.codeaurora.org/external/imx/linux-imx.git

The wait time is short that cause recording failed sometimes

Signed-off-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Reviewed-by: Sandor.yu <sandor.yu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../media/platform/imx8/hdmi/mxc-hdmi-rx-audio.c   |    7 +++----
 1 file changed, 3 insertions(+), 4 deletions(-)

diff --git a/drivers/media/platform/imx8/hdmi/mxc-hdmi-rx-audio.c b/drivers/media/platform/imx8/hdmi/mxc-hdmi-rx-audio.c
index d7d9a5e..22a55df 100644
--- a/drivers/media/platform/imx8/hdmi/mxc-hdmi-rx-audio.c
+++ b/drivers/media/platform/imx8/hdmi/mxc-hdmi-rx-audio.c
@@ -43,12 +43,11 @@ static int get_audio_infoframe(state_struct *state, unsigned int *chan)
 
 	do {
 		cdn_apb_read(state, ADDR_SINK_PIF + (PKT_INT_STATUS << 2), &regread);
-		udelay(10);
+		udelay(100);
 		times++;
-	} while (!(regread & (1 << 0)) && times < 100);
-
+	} while (!(regread & (1 << 0)) && times < 5000);
 
-	if (times == 100) {
+	if (times == 5000) {
 		ret = -EINVAL;
 		return ret;
 	}
-- 
1.7.9.5

